showcase/build/iaf_cond_alpha_neuron.cpp

```diff
@@ -0,0 +1,551 @@
+
+  P_.E_ex = 0;
+
+  /* generated by template
+   * org.nest.nestml.neuron.function.MemberInitialization*/
+
+  P_.E_in = ((-85.0));
+
+  /* generated by template
+   * org.nest.nestml.neuron.function.MemberInitialization*/
+
+  P_.E_L = ((-70.0));
+
+  /* generated by template
+   * org.nest.nestml.neuron.function.MemberInitialization*/
+
+  P_.tau_syn_ex = 0.2;
+
+  /* generated by template
+   * org.nest.nestml.neuron.function.MemberInitialization*/
+
+  P_.tau_syn_in = 2.0;
+
+  /* generated by template
+   * org.nest.nestml.neuron.function.MemberInitialization*/
+
+  P_.I_e = 0;
+
+  /* generated by template
+   * org.nest.nestml.neuron.function.MemberInitialization*/
+
+  P_.I_stim = 0;
+
+  /* generated by template
+   * org.nest.nestml.neuron.function.MemberInitialization*/
+
+  S_.y[State_::V_m] = get_E_L();
+
+  /* generated by template
+   * org.nest.nestml.neuron.function.MemberInitialization*/
+
+  S_.y[State_::g_in] = 0;
+
+  /* generated by template
+   * org.nest.nestml.neuron.function.MemberInitialization*/
+
+  S_.y[State_::g_ex] = 0;
+
+  /* generated by template
+   * org.nest.nestml.neuron.function.MemberInitialization*/
+
+  S_.y[State_::__D_g_in] = 0;
+
+  /* generated by template
+   * org.nest.nestml.neuron.function.MemberInitialization*/
+
+  S_.y[State_::__D_g_ex] = 0;
+}
+
+iaf_cond_alpha_neuron::iaf_cond_alpha_neuron(const iaf_cond_alpha_neuron &n)
+    : Archiving_Node(), P_(n.P_), S_(n.S_), B_(n.B_, *this) {}
+
+/* ----------------------------------------------------------------
+* Destructors
+* ---------------------------------------------------------------- */
+
+iaf_cond_alpha_neuron::~iaf_cond_alpha_neuron() {
+  // GSL structs may not have been allocated, so we need to protect destruction
+  if (B_.__s)
+    gsl_odeiv_step_free(B_.__s);
+  if (B_.__c)
+    gsl_odeiv_control_free(B_.__c);
+  if (B_.__e)
+    gsl_odeiv_evolve_free(B_.__e);
+}
+
+/* ----------------------------------------------------------------
+* Node initialization functions
+* ---------------------------------------------------------------- */
+
+void iaf_cond_alpha_neuron::init_state_(
+    const Node &proto) { // TODO inner components
+
+  const iaf_cond_alpha_neuron &pr = downcast<iaf_cond_alpha_neuron>(proto);
+  S_ = pr.S_;
+}
+
+/* generated by template
+ * org.nest.nestml.neuron.function.GSLDifferentiationFunction*/
+extern "C" inline int iaf_cond_alpha_neuron_dynamics(double, const double y[],
+                                                     double f[], void *pnode) {
+  typedef iaf_cond_alpha_neuron::State_ State_;
+  // get access to node so we can almost work as in a member function
+  assert(pnode);
+  const iaf_cond_alpha_neuron &node =
+      *(reinterpret_cast<iaf_cond_alpha_neuron *>(pnode));
+
+  // y[] here is---and must be---the state vector supplied by the integrator,
+  // not the state vector in the node, node.S_.y[].
+
+  double I_syn_exc = y[State_::g_ex] * (y[State_::V_m] - node.get_E_ex());
```

Comment:
I think it would be nice to declare intermediate variables to avoid the getters inside the equations.
If some variables are use several times, it would also avoid multiple calls to get in this critical function.